src/Hng.Web/Repo/GenericRepository.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Hng.Infrastructure.Context;
+using Hng.Web.Repo.Interface;
+using Microsoft.EntityFrameworkCore;
+
+namespace Hng.Web.Repo
+{
+    public class GenericRepository<T> : IGenericRepository<T> where T : class
```

Comment:
@davideshett You should have a base class called EntityBase that has a single Guid Id property. All other entities must then inherit this base class. So that you can now have IGenericRepository<T> where T : EntityBase.

That is the right approach. With this I can get IGenericRepository<String> which makes no sense.